### ISSUE TITLE
Broken parsing x AST repr roundtrip

### ISSUE THREAD

**Issue by notdanilo** [enso-org/ide#296]
### General Summary
The Parser isn't parsing empty lines properly.

### Steps to Reproduce
1. checkout `wip/dg/broken-parser-ast-roundtrip` of https://github.com/luna/ide
2. run `wasm-pack test --chrome src/rust/ide/parser/`

### Expected Result
Test passes.

### Actual Result
Test fails.

### Enso Version
`wip/dg/broken-parser-ast-roundtrip` of https://github.com/luna/ide, branched out from `e4a9d7742a4e4d4d0620b5ffe1fd60261438adb2`

**Comment by farmaazon** [comment]
The branch is not on the remote anymore. Can someone tell how to reproduce the issue?

**Comment by mwu-tow** [comment]
The repro was:
```rust
#[cfg(test)]
mod tests {
    use ast::HasRepr;
    use super::Parser;
    use super::api::IsParser;
    use wasm_bindgen_test::wasm_bindgen_test;

    wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);

    fn roundtrip_program(program:&str) {
        let mut parser = Parser::new_or_panic();
        let ast        = parser.parse(program.to_string(), Default::default()).unwrap();
        assert_eq!(ast.repr(), program, "{:#?}", ast);
    }

    #[wasm_bindgen_test]
    fn roundtrip() {
        let programs = vec![
            "main = 10 + 10",
            "main =\n    a = 10\n    b = 20\n    a * b",
            "main =\n    foo a =\n        a * 10\n    foo 10\n    print \"hello\"",
            "main =\n    foo\n    \n    bar",
            "main =\n    \n    foo\n    \n    bar"
        ];
        for program in programs {
            roundtrip_program(program);
        }
    }
}
```

**Comment by mwu-tow** [comment]
Just checked and this seems to be already resolved in 71a6b2122135692686bfa1cff62655c8c3def68c
